plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/SnapshotsTable.java

```diff
@@ -81,29 +87,39 @@ private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Conne
     {
         PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
 
+        TableScan tableScan = buildTableScan(icebergTable, SNAPSHOTS);
         TimeZoneKey timeZoneKey = session.getTimeZoneKey();
-        icebergTable.snapshots().forEach(snapshot -> {
-            pagesBuilder.beginRow();
-            pagesBuilder.appendTimestampTzMillis(snapshot.timestampMillis(), timeZoneKey);
-            pagesBuilder.appendBigint(snapshot.snapshotId());
-            if (checkNonNull(snapshot.parentId(), pagesBuilder)) {
-                pagesBuilder.appendBigint(snapshot.parentId());
-            }
-            if (checkNonNull(snapshot.operation(), pagesBuilder)) {
-                pagesBuilder.appendVarchar(snapshot.operation());
-            }
-            if (checkNonNull(snapshot.manifestListLocation(), pagesBuilder)) {
-                pagesBuilder.appendVarchar(snapshot.manifestListLocation());
-            }
-            if (checkNonNull(snapshot.summary(), pagesBuilder)) {
-                pagesBuilder.appendVarcharVarcharMap(snapshot.summary());
-            }
-            pagesBuilder.endRow();
+
+        tableScan.planFiles().forEach(fileScanTask -> {
+            DataTask dataTask = (DataTask) fileScanTask;
+            dataTask.rows().forEach(structLike -> addRow(pagesBuilder, structLike, timeZoneKey));
         });
 
         return pagesBuilder.build();
     }
 
+    private static void addRow(PageListBuilder pagesBuilder, StructLike structLike, TimeZoneKey timeZoneKey)
+    {
+        pagesBuilder.beginRow();
+        pagesBuilder.appendTimestampTzMillis(structLike.get(0, Long.class), timeZoneKey);
```

Comment:
Can we infer these indices from the `SnapshotsTable#schema` rather than relying on ordering?

Reply:
> any changes to the underlying Iceberg's metadata table can be used here

If you want to do that, we should also use the column names and types from the Table's schema, rather than having a fixed set of columns in the Page.

Reply:
@alexjo2144 yes, I thought about that,
but then I wanted to start with using the `iceberg_metadata_table.scan()` so that we don't need to have the logic of reading the data from these scans.

Then in the 2nd step can work on the schema part as well.

Reply:
Yea I kind of agree, I think we could use snapshotsTable.schema, and then use it to find the map of field_name -> index of various fields.  It's more guarantee from Iceberg that field names don't change , positions change not that rarely:  https://github.com/apache/iceberg/pull/3015#issuecomment-951133690   

As we could use it for all metadata tables, it'd probably be better if there's some central Util or framework to convert Iceberg Schema to position, not sure if Trino has that already , like for populating regular data rows.

Reply:
thanks @szehon-ho let me check that part.

Reply:
@szehon-ho 

did a quick test:

1. we can get the fieldId using the columnName but that is not the index of the column ( fieldId = index +1, but can we rely on this?)

```
String committedAtColumnName = "committed_at";

schema.findField(committedAtColumnName).fieldId()
```

2.  I see a way to find the index directly, by using the index of the entries returned by `schema.columns()`.

```
List<NestedField> columns = schema.columns();

Map<String, Integer> columnNameToIndex = columns.stream()
                .collect(Collectors.toMap(NestedField::name, x -> index.getAndIncrement()));

```

I see the `schema.columns()` is being populated based finally here:
https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/Types.java#L544-L550

looks like, based on this the `fieldId-1` can also be used as the index of the column.
Or use the index in the list.

WDYT?

Reply:
Yep, its not fieldId exactly, I dont think you can rely on that, but map of name -> position.  I think its the code that you have  on #2 (assuming index is an integer starting at 0?)

Reply:
yes @szehon-ho its an `array list` and index will be the index of that name in the array list.

Reply:
@alexjo2144 and @szehon-ho updated the PR to use the schemaTable's columnName to find the position of the column.

Reply:
> Then in the 2nd step can work on the schema part as well.

Do you mean changing results of `$snapshots` automatically? For instance, if Iceberg library adds a new field, `$snapshots` returns it.